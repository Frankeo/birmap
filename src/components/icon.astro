---
interface Props {
  iconClass: string;
  value?: number | boolean;
}

const { iconClass, value } = Astro.props;
const isNumeric = !isNaN(Number(value?.toString()));
---

{
  (value != null && isNumeric) || value === true ? (
    <span
      class={`icon has-text-primary ${isNumeric ? "is-large" : "is-medium"}`}
    >
      {isNumeric ? <span>{value}</span> : ""}
      <i class={`${iconClass} fa-xl`} />
    </span>
  ) : (
    ""
  )
}
