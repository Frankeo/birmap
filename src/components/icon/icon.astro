---
import { getIconHtml } from "./icon-html";
interface Props {
  iconClass: string;
  value?: number | undefined;
}

const { iconClass, value } = Astro.props;
const htmlText = getIconHtml(iconClass, value);
---

<span set:html={htmlText} />
