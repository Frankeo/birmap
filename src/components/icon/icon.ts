export const getIconHtml = (
  iconClass: string,
  value?: number | undefined,
  isSmall: boolean = false,
): string => {
  const isNumeric = !isNaN(Number(value?.toString()));
  return /* html */ `<span
  style="${isSmall ? "margin: 0;" : "margin: 0 0.25rem;"}"
  class="icon-text has-text-primary has-text-weight-bold"
>
  ${isNumeric ? `<span>${value}</span>` : ""}
  <span class="icon">
    <i class="${iconClass} ${isSmall ? "" : "fa-xl"}"></i>
  </span>
</span>`;
};
