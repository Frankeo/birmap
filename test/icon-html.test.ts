import { getIconHtml } from "../src/components/icon/icon-html";
import { expect, test, describe } from "vitest";

describe("Test Suite for getIconHtml", () => {
  const iconCss = "CSS";
  const value = 12;
  test("getIconHtml for boolean features", () => {
    const iconHtml = getIconHtml(iconCss);
    expect(iconHtml).toMatchSnapshot();
    expect(iconHtml).toContain("fa-xl");
    expect(iconHtml).toContain("margin: 0 0.25rem;");
  });
  test("getIconHtml for numeric features", () => {
    const iconHtml = getIconHtml(iconCss, value);
    expect(iconHtml).toMatchSnapshot();
    expect(iconHtml).toContain("fa-xl");
    expect(iconHtml).toContain(value);
    expect(iconHtml).toContain("margin: 0 0.25rem;");
  });
  test("getIconHtml for boolean features and small icons", () => {
    const iconHtml = getIconHtml(iconCss, undefined, true);
    expect(iconHtml).toMatchSnapshot();
    expect(iconHtml).not.toContain("fa-xl");
    expect(iconHtml).toContain("margin: 0;");
  });
  test("getIconHtml for numeric features and small icons", () => {
    const iconHtml = getIconHtml(iconCss, value, true);
    expect(iconHtml).toMatchSnapshot();
    expect(iconHtml).not.toContain("fa-xl");
    expect(iconHtml).toContain(value);
    expect(iconHtml).toContain("margin: 0;");
  });
});
