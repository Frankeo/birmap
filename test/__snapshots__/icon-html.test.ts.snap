// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test Suite for getIconHtml > getIconHtml for boolean features 1`] = `
"<span
  style=\\"margin: 0 0.25rem;\\"
  class=\\"icon-text has-text-primary has-text-weight-bold\\"
>
  
  <span class=\\"icon\\">
    <i class=\\"CSS fa-xl\\"></i>
  </span>
</span>"
`;

exports[`Test Suite for getIconHtml > getIconHtml for boolean features and small icons 1`] = `
"<span
  style=\\"margin: 0;\\"
  class=\\"icon-text has-text-primary has-text-weight-bold\\"
>
  
  <span class=\\"icon\\">
    <i class=\\"CSS \\"></i>
  </span>
</span>"
`;

exports[`Test Suite for getIconHtml > getIconHtml for numeric features 1`] = `
"<span
  style=\\"margin: 0 0.25rem;\\"
  class=\\"icon-text has-text-primary has-text-weight-bold\\"
>
  <span>12</span>
  <span class=\\"icon\\">
    <i class=\\"CSS fa-xl\\"></i>
  </span>
</span>"
`;

exports[`Test Suite for getIconHtml > getIconHtml for numeric features and small icons 1`] = `
"<span
  style=\\"margin: 0;\\"
  class=\\"icon-text has-text-primary has-text-weight-bold\\"
>
  <span>12</span>
  <span class=\\"icon\\">
    <i class=\\"CSS \\"></i>
  </span>
</span>"
`;
